packages/use-intl/src/core/createFormatter.tsx

```diff
@@ -284,7 +296,11 @@ export default function createFormatter({
   type FormattableListValue = string | ReactElement;
   function list<Value extends FormattableListValue>(
     value: Iterable<Value>,
-    formatOrOptions?: string | Intl.ListFormatOptions
+    formatOrOptions?:
+      | (keyof IntlFormats['list'] extends string
```

Comment:
For code readibility I tried to create a type helper for this like that:

```
type GlobalFormat<T extends keyof Formats> =
    keyof IntlFormats[T] extends string ? keyof IntlFormats[T] : string;
```

So then we could do
```
formatOrOptions?: GlobalFormat<"dateTime"> | Intl.ListFormatOptions
```

But I don't know why, it's not working, so I just caved and placed this verbose thing for now